### DISCUSSION TITLE
inquiry about misorientation axis calculation

### DISCUSSION THREAD

**Discussion by jdkang0810** [mtex-toolbox/mtex#2245] [Ask Anything] [answered]
I have come across an issue during my work.
I am trying to calculate the misorientation angles and axes for each grain boundary in hexagonal crystals using MTEX.
While MTEX seems to compute the misorientation angles correctly, there is a discrepancy in the misorientation axes represented by miller index between MTEX and Aztec Crystal.
For example, for the same boundary, Aztec Crystal provides a misorientation axis with a Miller index of <-1 -1 2 0>, while MTEX gives different value.
I suspect there might be some additional considerations regarding axis rotation or coordinate system differences.

1. Identify the critical boundary.
I think this boundary can be seen in an EBSD image with a difference angle between these grains greater than 10 degrees.
I attached EBSD image.
![2024-10-12 22_28_30 449](https://github.com/user-attachments/assets/0cd71739-c218-4cdb-9f62-d7828b1aca1e)

2. Identify the grain/EBSD orientations on both sides of the boundary.
The Euler angles for each grain are quite consistent between the two programs.
In Aztec Crystal, the Euler angles for Grain 1 are approximately 117.32°, 54.54°, and 53.01°, and for Grain 2 they are approximately 138.65°, 110.9°, and 52.23°.
In MTEX, the average Euler angles for grain 1 are 117.18°, 54.60°, and 52.91°, and for grain 2 they are 138.52°, 110.97°, and 52.15°, respectively.
So there seems to be no significant difference in the Euler angle inputs between the two programs.  

When importing a ctf file, is it necessary to transform the coordinate system from Aztec Crystal?
If so, could you explain what information is needed to perform this transformation?

Below is the part of the code used to import the data into MTEX.
CS = crystalSymmetry('6/mmm', [2.95 2.95 4.73], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Ti-Hex', 'color', [0.53 0.81 0.98]);
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane')
ebsd = EBSD.load(fname,CS,'interface','ctf','convertSpatial2EulerReferenceFrame');

3-4. calculate the misorientation and compare it to boundary.misorientation and calculate the axis of misorientation
Using the point-to-point disorientation function (with 11 pixel width) in Aztec Crystal, the disorientation angle comes out to be 59.87 degrees, and the axis is <h k i l> = <-1 -1 2 0>. 
This value did not change with different pixel widths.
On the other hand, when calculating the misorientation angle for the same boundary in MTEX, the average is 60.05° and the axis is <h k i l> = <-1.51 2.95 -1.44 0.12>.
Additionally, when calculating based on the average Euler angles of each grain, the misorientation angle is 59.94° , and the axis is <h k i l> =  <2.60 -2.18 -0.42 -1.54>.

There is a significant difference between the axis values in Aztec Crystal and MTEX, and even in MTEX the results varied depending on how the axis was calculated. 
Is there a way to solve the problem where the misorientation axis is different depending on the calculation method?

I attach the code I used to get these values in MTEX.
```matlab
a1 = SelectedBoundaries.misorientation;
a1_angle = angle(a1);
a1_angle_mean_deg = rad2deg(mean(a1_angle));
a1_axis = axis(SelectedBoundaries.misorientation)
a1_axis_mean = mean(a1);

meanOri1 = grains.meanOrientation(selected_grain_pairs(i, 1))  
meanOri2 = grains.meanOrientation(selected_grain_pairs(i, 2));
angle(meanOri1, meanOri2) ./ degrees

axis(meanOri1, meanOri2)
Miller(ans, CS)
```

**Comment by ralfHielscher** [comment, marked as answer]
In your case the misorientation axes of Aztec Crystal and MTEX coincide. They only differ by (1) rounding and (2) which of the symmetrically equivalent axes is returned. Check out the following code 
```matlab
% set up the misorientation
cs = crystalSymmetry('6/mmm', [2.95 2.95 4.73], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Ti-Hex', 'color', [0.53 0.81 0.98]);

ori1 = orientation.byEuler(117.1828*degree,54.6*degree,52.9143*degree,cs)
ori2 = orientation.byEuler(138.5187*degree,110.9740*degree,52.1452*degree,cs)

angle(ori1,ori2)/degree

% compute the misorientation
mori = inv(ori1)*ori2;

% compute the misorientation axis
ax1 = mori1.axis

% round the misorientation axis - now it coincides with the AZTEC axis
ax2 = round(mori.axis,'maxHKL',3)

% the angle between both axes
angle(ax1,ax2) /degree

% show both axes in an ipf plot
plot(mori.axis,'fundamentalSector')
hold on
plot(round(mori.axis,'maxHHK',3))
hold off
```
Ralf.
